optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/incremental/IncrementalScoreDirector.java

```diff
@@ -112,25 +112,25 @@ public boolean isConstraintMatchEnabled() {
     }
 
     @Override
-    public Map<Object, Indictment> getIndictmentMap() {
+    public <Score_ extends Score<Score_>> Map<Object, Indictment<Score_>> getIndictmentMap() {
         if (!isConstraintMatchEnabled()) {
             throw new IllegalStateException("When constraintMatchEnabled (" + isConstraintMatchEnabled()
                     + ") is disabled in the constructor, this method should not be called.");
         }
         Map<Object, Indictment> incrementalIndictmentMap =
                 ((ConstraintMatchAwareIncrementalScoreCalculator<Solution_>) incrementalScoreCalculator).getIndictmentMap();
```

Comment:
If we're BC breaking ScoreManager for justification API gain, we should also do it to ConstraintMatchAwareIncrementalScoreCalculator, which one one uses and isn't part of the api package. So I am doing that now :)